dapp/src/client/scripts/app.jsx

```diff
@@ -152,9 +148,16 @@ export class App extends React.Component {
 
   buyEnergy(contractAddress) {
     const contract = makeContract(contractAddress);
+    var index = 0;
+    for (var i = 0; i < this.state.contracts.length; i++) {
```

Comment:
The above means we have to calculate the index into this array when we want to update the tx field